tests/runtimes/test_application.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pathlib
+
+import pytest
+
+import mlrun
+
+
+def test_create_application_runtime():
+    fn: mlrun.runtimes.ApplicationRuntime = mlrun.code_to_function(
+        "application-test", kind="application", image="mlrun/mlrun"
+    )
+    assert fn.kind == mlrun.runtimes.RuntimeKinds.application
+    assert fn.spec.image == "mlrun/mlrun"
+    assert fn.metadata.name == "application-test"
+    # base64 prefix of the reverse proxy code
+    assert (
+        "Ly8gQ29weXJpZ2h0IDIwMjQgSWd1YXppbwovLwovLyBMaWN"
+        in fn.spec.build.functionSourceCode
+    )
```

Comment:
Thinking about it again, this is not very resilient to changes in the reverse proxy source code, let alone if you are reusing it in other tests.
Better to automate it (using the `filepath` below)